src/App.jsx

```diff
@@ -14,33 +16,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const initialGoods = goodsFromServer;
+  const [goods, setGoods] = useState([...initialGoods]);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortBy, setSortBy] = useState('');
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const updateGoods = () => {
+    const newGoods = [...initialGoods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    if (sortBy === 'alphabet') {
+      newGoods.sort((a, b) => a.localeCompare(b));
+    } else if (sortBy === 'length') {
+      newGoods.sort((a, b) => a.length - b.length);
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (isReversed) {
+      newGoods.reverse();
+    }
+
+    setGoods(newGoods);
+  };
+
+  useEffect(() => {
+    updateGoods();
+  }, [sortBy, isReversed]);
+
+  const handleSortAlphabetically = () => {
+    setSortBy('alphabet');
+  };
+
+  const handleSortByLength = () => {
+    setSortBy('length');
+  };
+
+  const handleReverse = () => {
+    setIsReversed(!isReversed);
```

Comment:
```suggestion
    setIsReversed(prev => !prev);
```
When you need to set state depending on the previous one use functional setState